linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -175,6 +175,8 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 			foreach (TypeReference tr in assembly.MainModule.GetTypeReferences ()) {
 				if (hash.ContainsKey (tr))
 					continue;
+				if (!CanChangeScope (tr))
```

Comment:
So the reason why this added added was because we had this piece of code instead of that condition:

`if (type.IsWindowsRuntimeProjection) continue;`

Michael assumed that you don't want any of the "Windows Runtime stuff" in upstream so he factored it out into a virtual method that we now overwrite. I think the cleanest solution would be to just keep the original condition to continue if a type is a Windows Runtime projection.

The reason why we cannot change scope when a type is windows runtime projection is because cecil just throws if you try to do that (after all, a projection is not a real reference):

https://github.com/jbevain/cecil/blob/master/Mono.Cecil/TypeReference.cs#L145